pkg/api/grpc/grpc.go

```diff
@@ -122,29 +126,37 @@ func NewAPI(opts APIOpts) API {
 // validateAndGetPubsbuAndTopic validates the request parameters and returns the pubsub interface, pubsub name, topic name, rawPayload metadata if set
 // or an error.
 func (a *api) validateAndGetPubsubAndTopic(pubsubName, topic string, reqMeta map[string]string) (pubsub.PubSub, string, string, bool, error) {
+	var err error
+	pubsubType := string(contribMetadata.PubSubType)
+
 	if a.pubsubAdapter == nil {
-		return nil, "", "", false, status.Error(codes.FailedPrecondition, messages.ErrPubsubNotConfigured)
+		err = apierrors.NotConfigured(pubsubName, pubsubType, reqMeta, codes.FailedPrecondition, http.StatusBadRequest, "ERR_PUBSUB_NOT_CONFIGURED", kiterrors.CodePrefixPubSub+kiterrors.CodeNotConfigured)
+		return nil, "", "", false, err
 	}
 
 	if pubsubName == "" {
-		return nil, "", "", false, status.Error(codes.InvalidArgument, messages.ErrPubsubEmpty)
+		err = apierrors.PubSubNameEmpty(pubsubName, pubsubType, reqMeta)
+		return nil, "", "", false, err
 	}
 
 	thepubsub, ok := a.Universal.CompStore().GetPubSub(pubsubName)
 	if !ok {
-		return nil, "", "", false, status.Errorf(codes.InvalidArgument, messages.ErrPubsubNotFound, pubsubName)
+		err = apierrors.NotFound(pubsubName, pubsubType, reqMeta, codes.InvalidArgument, http.StatusNotFound, "ERR_PUBSUB_NOT_FOUND", kiterrors.CodePrefixPubSub+kiterrors.CodeNotFound)
+		return nil, "", "", false, err
 	}
 
 	if topic == "" {
-		return nil, "", "", false, status.Errorf(codes.InvalidArgument, messages.ErrTopicEmpty, pubsubName)
+		err = apierrors.PubSubTopicEmpty(pubsubName, pubsubType, reqMeta)
+		return nil, "", "", false, err
 	}
 
 	rawPayload, metaErr := contribMetadata.IsRawPayload(reqMeta)
 	if metaErr != nil {
-		return nil, "", "", false, messages.ErrPubSubMetadataDeserialize.WithFormat(metaErr)
+		err = apierrors.PubSubMetadataDeserialize(pubsubName, pubsubType, reqMeta, metaErr)
+		return nil, "", "", false, err
 	}
 
-	return thepubsub.Component, pubsubName, topic, rawPayload, nil
+	return thepubsub.Component, pubsubName, topic, rawPayload, err
```

Comment:
```suggestion
	return thepubsub.Component, pubsubName, topic, rawPayload, nil
```

Reply:
if err is nil, it will return nil here anyways.